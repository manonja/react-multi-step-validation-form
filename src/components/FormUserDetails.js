import React from 'react';

const FormUserDetails = ( {values, handleChange, nextStep} ) => {

  return (
    <div className="wrapper">
      <div className="form-wrapper">
        <div className='page-detail'>
          <h3 className="alignleft">User Details</h3>
          <p className="aligncenter">Privacy</p>
          <p className="alignright">Done</p>
        </div>
        <form noValidate className="form">
          <div className="name">
            <input
              id="input-form-name"
              className={values.formErrorsMessages.name.length > 0 ? "error" : null}
              placeholder="Name *"
              autoComplete="current-name"
              type="text"
              name="name"
              noValidate
              onChange={handleChange}
              defaultValue={values.name}
            />
            {values.formErrorsMessages.name.length > 0 && (
              <span className="errorMessage">{values.formErrorsMessages.name}</span>
            )}
          </div>
          <div className="role">
            <input
              id="input-form-role"
              placeholder="Role"
              autoComplete="current-role"
              type="text"
              name="role"
              onChange={handleChange}
              defaultValue={values.role}
            />
          </div>
          <div className="email">
            <input
              id="input-form-email"
              className={values.formErrorsMessages.email.length > 0 ? "error" : null}
              placeholder="Email *"
              autoComplete="current-email"
              type="email"
              name="email"
              onChange={handleChange}
              defaultValue={values.email}
            />
            {values.formErrorsMessages.email.length > 0 && (
              <span className="errorMessage">{values.formErrorsMessages.email}</span>
            )}
          </div>
          <div className="password">
            <input
              id="input-form-password"
              className={values.formErrorsMessages.password.length > 0 ? "error" : null}
              placeholder="Password *"
              autoComplete="current-password"
              type="password"
              name="password"
              onChange={handleChange}
              defaultValue={values.password}
            />
            {values.formErrorsMessages.password.length > 0 && (
              <span className="errorMessage">{values.formErrorsMessages.password}</span>
            )}
          </div>
          <button 
            type="submit" 
            id='btn-next' 
            onClick={nextStep}
          >
            Next
          </button>
        </form>
        <br/>
        <small><i>Fields marked with * are required</i></small>
      </div>
    </div>  
  )
};
    

export default FormUserDetails;
