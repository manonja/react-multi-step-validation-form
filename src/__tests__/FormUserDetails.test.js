
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import FormUserDetails from '../components/FormUserDetails';

let wrapper;

beforeEach(() => {
    wrapper = shallow(
      <FormUserDetails 
          values={{name: '', role: '', email: '', password: '', formErrorsMessages: { name: '', role: '', email: '', password: ''}}} 
          handleChange={() => {}} 
          nextStep={() => {}} 
      />
    )
});

describe('<ForUserDetails /> rendering', () => {

    it('ForUserDetails: renders correctly', () => {
        const tree = renderer.create(
            <FormUserDetails 
                values={{name: '', role: '', email: '', password: '', formErrorsMessages: { name: '', role: '', email: '', password: ''}}} 
                handleChange={() => {}} 
                nextStep={() => {}} 
            />).toJSON() 
        
        expect(tree).toMatchSnapshot();
    });

    it('renders without crashing', () => {
        expect(wrapper)
    });
  
    it('should render a <form /> to type the informations', () => {
        expect(wrapper.find('form').length).toEqual(1);      
    });
});

describe('<FormUserDetails /> interactions', () => {

    it('should call the onClick function when \'Next\' button is clicked', () => { 
        const nextStep = jest.fn();
        const handleChange = jest.fn();
        const handleSubmit = jest.fn();

        const wrapper = mount(
            <FormUserDetails 
                values={{name: '', role: '', email: '', password: '', formErrorsMessages: { name: '', role: '', email: '', password: ''}}} 
                handleChange={handleChange} 
                nextStep={nextStep}
            />
        );
        wrapper.find('button').at(0).simulate('click');
        expect(nextStep).toHaveBeenCalled();
    });
});




