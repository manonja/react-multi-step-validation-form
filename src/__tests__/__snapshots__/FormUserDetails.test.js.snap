// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForUserDetails /> rendering ForUserDetails: renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="form-wrapper"
  >
    <div
      className="page-detail"
    >
      <h3
        className="alignleft"
      >
        User Details
      </h3>
      <p
        className="aligncenter"
      >
        Privacy
      </p>
      <p
        className="alignright"
      >
        Done
      </p>
    </div>
    <form
      className="form"
      noValidate={true}
    >
      <div
        className="name"
      >
        <input
          autoComplete="current-name"
          className={null}
          defaultValue=""
          id="input-form-name"
          name="name"
          noValidate={true}
          onChange={[Function]}
          placeholder="Name *"
          type="text"
        />
      </div>
      <div
        className="role"
      >
        <input
          autoComplete="current-role"
          defaultValue=""
          id="input-form-role"
          name="role"
          onChange={[Function]}
          placeholder="Role"
          type="text"
        />
      </div>
      <div
        className="email"
      >
        <input
          autoComplete="current-email"
          className={null}
          defaultValue=""
          id="input-form-email"
          name="email"
          onChange={[Function]}
          placeholder="Email *"
          type="email"
        />
      </div>
      <div
        className="password"
      >
        <input
          autoComplete="current-password"
          className={null}
          defaultValue=""
          id="input-form-password"
          name="password"
          onChange={[Function]}
          placeholder="Password *"
          type="password"
        />
      </div>
      <button
        id="btn-next"
        onClick={[Function]}
        type="submit"
      >
        Next
      </button>
    </form>
    <br />
    <small>
      <i>
        Fields marked with * are required
      </i>
    </small>
  </div>
</div>
`;
